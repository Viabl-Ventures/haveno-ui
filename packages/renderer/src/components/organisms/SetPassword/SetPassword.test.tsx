// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { describe, expect, it, vi } from "vitest";
import { fireEvent, render } from "@testing-library/react";
import { screen } from "@testing-library/dom";
// import userEvent from "@testing-library/user-event";
import { AppProviders } from "@atoms/AppProviders";
import { SetPassword } from ".";

describe("organisms::SetPassword", () => {
  it("renders without exploding", () => {
    const onBackSpy = vi.fn();
    const onNextSpy = vi.fn();
    const { asFragment, unmount } = render(
      <AppProviders>
        <SetPassword value="" onGoBack={onBackSpy} onNext={onNextSpy} />
      </AppProviders>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it("calls onGoBack", async () => {
    const onBackSpy = vi.fn();
    const onNextSpy = vi.fn();
    const { unmount } = render(
      <AppProviders>
        <SetPassword value="" onGoBack={onBackSpy} onNext={onNextSpy} />
      </AppProviders>
    );
    expect(onBackSpy).to.not.toHaveBeenCalled();
    fireEvent.click(await screen.findByLabelText("Click to go back"));
    expect(onBackSpy).to.toHaveBeenCalledTimes(1);
    expect(onNextSpy).to.not.toHaveBeenCalled();
    unmount();
  });

  it("blocks submit if validation fails", async () => {
    const onBackSpy = vi.fn();
    const onNextSpy = vi.fn();

    const { unmount } = render(
      <AppProviders>
        <SetPassword value="" onGoBack={onBackSpy} onNext={onNextSpy} />
      </AppProviders>
    );
    expect(onNextSpy).to.not.toHaveBeenCalled();
    fireEvent.click(await screen.findByLabelText("Click to submit"));
    expect(onNextSpy).to.not.toHaveBeenCalled();
    unmount();
  });

  it("calls onSubmit if validation succeeds", async () => {
    const onBackSpy = vi.fn();
    const onNextSpy = vi.fn();
    // const user = userEvent.setup();
    const { unmount, getByLabelText } = render(
      <AppProviders>
        <SetPassword value="" onGoBack={onBackSpy} onNext={onNextSpy} />
      </AppProviders>
    );
    // await user.click(getByLabelText("Enter password"));
    // await user.keyboard("Qwe$9999{Tab}Qwe$9999");

    // expect(onNextSpy).to.not.toHaveBeenCalled();
    fireEvent.click(getByLabelText("Enter password"));
    fireEvent.input(getByLabelText("Enter password"), {
      currentTarget: {
        value: "Qwe$9999",
      },
      target: {
        value: "Qwe$9999",
      },
    });
    await new Promise((resolve) => setTimeout(resolve, 1000));
    fireEvent.blur(getByLabelText("Enter password"));
    // fireEvent.focus(getByLabelText("Repeat password"));
    // fireEvent.blur(getByLabelText("Enter password"));
    // await new Promise((resolve) => setTimeout(resolve, 1000));
    // await new Promise((resolve) => setTimeout(resolve, 1000));

    fireEvent.submit(getByLabelText("Click to submit"));
    // expect(onNextSpy).to.toHaveBeenCalledTimes(1);
    unmount();
  });
});
