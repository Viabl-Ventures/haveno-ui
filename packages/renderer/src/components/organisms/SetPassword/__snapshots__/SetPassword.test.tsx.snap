// Vitest Snapshot v1

exports[`organisms::SetPassword > renders without exploding 1`] = `
<DocumentFragment>
  <form
    data-testid="change-password-form"
  >
    <div
      class="mantine-Stack-root mantine-lfk3cq"
    >
      <div
        class="mantine-Container-root mantine-me1cdz"
      >
        <h1
          class="mantine-Title-root mantine-1rawl7h"
        >
          Create password
        </h1>
      </div>
      <div
        class="mantine-Text-root mantine-1ourfup"
      >
        All your data is stored locally on your machine. Haveno uses solely a password.
      </div>
      <div
        class="mantine-TextInput-root mantine-14qek68"
      >
        <label
          class="mantine-TextInput-label mantine-1bjo575"
          for="password"
          id="password-label"
        >
          Password
        </label>
        <div
          class="mantine-TextInput-wrapper mantine-12sbrde"
        >
          <input
            aria-invalid="false"
            aria-label="Enter password"
            class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-nk8491"
            id="password"
            tabindex="1"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="mantine-TextInput-root mantine-14qek68"
      >
        <label
          class="mantine-TextInput-label mantine-1bjo575"
          for="repeatPassword"
          id="repeatPassword-label"
        >
          Repeat password
        </label>
        <div
          class="mantine-TextInput-wrapper mantine-12sbrde"
        >
          <input
            aria-invalid="false"
            aria-label="Repeat password"
            class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-nk8491"
            id="repeatPassword"
            tabindex="2"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="mantine-63n06h"
      />
      <div
        class="mantine-Group-root mantine-19jxmdp"
      >
        <button
          aria-label="Click to go back"
          class="mantine-UnstyledButton-root mantine-Group-child mantine-1smyfmn"
          type="button"
        >
          <span
            class="mantine-Text-root mantine-1mwiqnz"
          >
            Go Back
          </span>
        </button>
        <button
          aria-label="Click to submit"
          class="mantine-Button-filled mantine-Button-root mantine-Group-child mantine-s2bdxi"
          type="submit"
        >
          <div
            class="mantine-3xbgk5 mantine-Button-inner"
          >
            <span
              class="mantine-qo1k2 mantine-Button-label"
            >
              Next
            </span>
          </div>
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
